Lib/logging/handlers.py

```diff
@@ -255,6 +255,7 @@ def __init__(self, filename, when='h', interval=1, backupCount=0,
             raise ValueError("Invalid rollover interval specified: %s" % self.when)
 
         self.extMatch = re.compile(self.extMatch, re.ASCII)
+        self.file_number = re.compile(r'(\()([0-9])(\))$')
```

Comment:
Any reason to limit to just one digit? Also, why do we need capturing groups for the parentheses?

Reply:
> Any reason to limit to just one digit?

no, this was an oversight on my part, thanks.

> Also, why do we need capturing groups for the parentheses?

Me from 10 days ago had a good reason I think :D. I think my thought process was having a group like 
`group = ['(', '12', ')']` would be easier to parse the number out as you can get the number from group[1] but the same effect could be had from `match[1:-1]` if the parenthesis weren't in a separate group. Funnily enough this situation would never arise thanks to the first issue you pointed out.

I'll look at this again when I get back from work to see If I've missed anything.